jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/openshift/OpenShiftBuildServiceUtils.java

```diff
@@ -0,0 +1,214 @@
+
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.LocalObjectReferenceBuilder;
+import io.fabric8.openshift.api.model.BuildConfig;
+import io.fabric8.openshift.api.model.BuildConfigSpec;
+import io.fabric8.openshift.api.model.BuildOutput;
+import io.fabric8.openshift.api.model.BuildOutputBuilder;
+import io.fabric8.openshift.api.model.BuildStrategy;
+import io.fabric8.openshift.api.model.BuildStrategyBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jkube.kit.build.api.assembly.ArchiverCustomizer;
+import org.eclipse.jkube.kit.common.util.IoUtil;
+import org.eclipse.jkube.kit.common.util.KubernetesHelper;
+import org.eclipse.jkube.kit.config.image.ImageConfiguration;
+import org.eclipse.jkube.kit.config.image.ImageName;
+import org.eclipse.jkube.kit.config.image.build.BuildConfiguration;
+import org.eclipse.jkube.kit.config.image.build.JKubeBuildStrategy;
+import org.eclipse.jkube.kit.config.service.BuildServiceConfig;
+import org.eclipse.jkube.kit.config.service.JKubeServiceException;
+import org.eclipse.jkube.kit.config.service.JKubeServiceHub;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.eclipse.jkube.kit.build.api.helper.BuildUtil.extractBaseFromDockerfile;
+import static org.eclipse.jkube.kit.config.service.openshift.ImageStreamService.resolveImageStreamName;
+import static org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.DEFAULT_BUILD_OUTPUT_KIND;
+import static org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.DEFAULT_S2I_BUILD_SUFFIX;
+import static org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.DOCKER_IMAGE;
+import static org.eclipse.jkube.kit.config.service.openshift.OpenshiftBuildService.IMAGE_STREAM_TAG;
+
+public class OpenShiftBuildServiceUtils {
+
+  private OpenShiftBuildServiceUtils() {}
+
+  protected static File createBuildArchive(JKubeServiceHub jKubeServiceHub, ImageConfiguration imageConfig) throws JKubeServiceException {
+    // Adding S2I artifacts such as environment variables in S2I mode
+    final ArchiverCustomizer customizer = createS2IArchiveCustomizer(jKubeServiceHub.getBuildServiceConfig(), imageConfig);
+    try {
+      return jKubeServiceHub.getDockerServiceHub().getArchiveService()
+          .createDockerBuildArchive(imageConfig, jKubeServiceHub.getConfiguration(), customizer);
+    } catch (IOException e) {
+      throw new JKubeServiceException("Unable to create the build archive", e);
+    }
+  }
+
+  /**
+   * Returns the applicable name for the S2I Build resource considering the provided {@link ImageName} and
+   * {@link BuildServiceConfig}.
+   */
+  protected static String computeS2IBuildName(BuildServiceConfig config, ImageName imageName) {
```

Comment:
Why did you choose methods in this class to be protected? I don't see this utility class getting inherited anywhere?

Reply:
Methods from OpenShiftBuildService were moved to this class one by one. So the reason is probably because in the initial iterations the method was consumed from OpenShiftBuildService and later on the calling methods were moved to the Utils class too.

I'll change this.